plugins/notifications-backend/src/database/DatabaseNotificationsStore.ts

```diff
@@ -87,24 +89,63 @@ export class DatabaseNotificationsStore implements NotificationsStore {
   };
 
   private mapToNotifications = (rows: any[]): Notification[] => {
-    return rows.map(row => ({
-      id: row.id,
-      user: row.user,
-      created: new Date(row.created),
-      saved: row.saved,
-      read: row.read,
-      updated: row.updated,
-      origin: row.origin,
-      payload: {
-        title: row.title,
-        description: row.description,
-        link: row.link,
-        topic: row.topic,
-        severity: row.severity,
-        scope: row.scope,
-        icon: row.icon,
-      },
-    }));
+    let order = 0;
+    const res = [
+      ...rows
+        .reduce((acc, row) => {
+          const metadata = row.name
+            ? {
+                name: row.name,
+                value: row.value,
+                type: row.type,
+              }
+            : undefined;
+          if (acc.has(row.id)) {
+            if (metadata) {
+              acc.get(row.id).notification.payload.metadata?.push(metadata);
+            }
+          } else {
+            order += 1;
+            acc.set(row.id, {
+              order,
+              notification: {
+                id: row.id,
+                user: row.user,
+                created: new Date(row.created),
+                saved: row.saved,
+                read: row.read,
+                updated: row.updated,
+                origin: row.origin,
+                payload: {
+                  title: row.title,
+                  description: row.description,
+                  link: row.link,
+                  topic: row.topic,
+                  severity: row.severity,
+                  scope: row.scope,
+                  icon: row.icon,
+                  ...(metadata && { metadata: [metadata] }),
+                },
+              },
+            });
+          }
+          return acc;
+        }, new Map<string, { order: number; notification: Notification }>())
+        .values(),
+    ] as { order: number; notification: Notification }[];
+
+    return res.sort((a, b) => a.order - b.order).map(e => e.notification);
+  };
+
+  private mapNotificationToMetadataDbRows = (notification: Notification) => {
+    return (
+      notification.payload?.metadata?.map(metadata => ({
+        notification_id: notification.id,
+        name_lower: metadata.name.toLowerCase(),
```

Comment:
i any toLower/UpperCase, it's worth adding an `'en-US'` argument to ensure that you don't get bitten by some crazy locales out there that don't do what you think they should be doing :) Some linters actually warn about this.

Reply:
Would be nice to configure eslint rule in Backstage for that 😛 😃

Reply:
@freben toLower/UpperCase doesn't take any parameters.
Are you suggesting to use `toLocaleLowerCase` instead? I see from here https://www.w3schools.com/jsref/jsref_tolocalelowercase.asp that the difference is very specific

Reply:
Right, yes.